packages-explorer/src/gui/result.js

```diff
@@ -0,0 +1,245 @@
+	switch (channel) {
+	case "nixos-unstable":
+		return "nixos/trunk-combined";
+	case "nixpkgs-unstable":
+		return "nixpkgs/trunk";
+	default:
+		return channel.replace(/^nixos-/, "nixos/release-");
+	}
+};
+
+const normalize_attrname = (name, channel) =>
+	channel.match(/^nixos-/) ? `nixpkgs.${name}` : name;
+
+const Platform = ({platform, attr, channel}) =>
+	<li key={platform}>
+		<a href={hydraLink(normalize_attrname(attr, channel), platform, channel_to_jobset(channel))}>
+			<tt>{platform}</tt>
+		</a>
+	</li>
+;
+
+/**
+ * Removes non-nixos platforms from nixos channels.
+ */
+const platforms_for_channel = (channel) =>
+	(platform) => !(channel.match(/^nixos-/) && platform.match(/-darwin$/));
+
+const Platforms = use(["channel"], [], ({channel, result: {attr, meta: {platforms}}}) =>
+	<tr>
+		<th>Platforms</th>
+		<td>
+			{
+				!platforms || platforms.length < 1 ? <NotSpecified />
+					: <ul class="platforms-list">
+						{
+							PLATFORMS
+								.filter(platforms_for_channel(channel))
+								.filter((platform) => platforms.indexOf(platform) > -1)
+								.map((platform) => <Platform key={platform} platform={platform} attr={attr} channel={channel} />)
+						}
+					</ul>
+			}
+		</td>
+	</tr>
+);
+
+const Homepage = ({result: {meta: {homepage}}}) =>
+	<tr>
+		<th>Homepage</th>
+		<td>
+			{
+				homepage && homepage.length > 0
+					? <a href={homepage} rel="nofollow">{homepage}</a>
+					: <NotSpecified />
+			}
+		</td>
+	</tr>
+;
+
+const License = ({result: {meta: {license}}}) =>
+	<tr>
+		<th>License</th>
+		<td>
+			{
+				license
+					? <FormattedLicense license={license} />
+					: <NotSpecified />
+			}
+		</td>
+	</tr>
+;
+
+const Maintainers = ({result: {meta: {maintainers}}}) =>
+	<tr>
+		<th>Maintainers</th>
+		<td>
+			{
+				maintainers && maintainers.length > 0
+					? <span>{
+						maintainers
+							.map((m) => `${m.name} <${m.email}>`)
+							.join(", ")
+					}</span>
+					: <NotSpecified />
+			}
+		</td>
+	</tr>
+;
+
+const LongDescription = ({result: {meta: {longDescription}}}) =>
+	<tr>
+		<th>Long description</th>
+		<td class="long-description pre">
+			{
+				longDescription && longDescription.length > 0
+					? <div>{longDescription}</div>
+					: <NotSpecified />
+			}
+		</td>
+	</tr>
+;
+
+const ROWS = [
+	Install,
+	Unfree,
+	Position,
+	Platforms,
+	Homepage,
+	License,
+	Maintainers,
+	LongDescription,
+];
+
+const ResultDetails = ({
+	result,
+	result: {attr},
+	even,
+	selected,
+}) =>
+	<tr
+		key="details"
+		class={[
+			"details",
+			even ? "even" : "odd",
+			selected === attr ? "is-selected" : "is-hidden",
+			isUnfree(result["meta"]["license"]) ? "is-unfree" : "is-free",
+		].join(" ")}
+	>
+		<td colspan={3}>
+			<div class="search-details">
+				<table>
+					<tbody>
+						{ROWS.map((Row, i) => <Row result={result} key={i} />)}
+					</tbody>
+				</table>
+				<div class="result--permalink">
+					<Link merge={true} state={{attr}}>
+						Link to <tt>{attr}</tt>
```

Comment:
I really like this, but was confused (honestly), and until I inspected the link, I wasn't sure what it did. I wondered:

 - is it taking me to the package's homepage?
 - is it a link to the hydra job?
 - is it taking me to the expression?
 - is it a permalink to this entry?
 - is it going to copy the permalink to this entry automatically?

I think the button is probably a bit big for what it actually is (a permalink to the package), and maybe the text should just say "permalink". What do you think?

_side note, I imagine it is tricky, but having it auto-copy the URL would be fancypants._

Reply:
As we make sure not to thrash the user's history, I think it is better not to thrash the user's clipboard unexpectedly. Users can right click and copy link location, or follow the link and copy what is in their location bar. Copying a permalink to the clipboard automatically started as a poor workaround for web application unable to handle proper back/forward navigation events.